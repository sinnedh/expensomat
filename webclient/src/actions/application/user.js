export const setToken = (token) => {
  return {
    type: 'APPLICATION:SET_TOKEN',
    token,
  }
}

export const setUser = (user) => {
  return {
    type: 'APPLICATION:SET_USER',
    user,
  }
}

export const resetUser = () => {
  return {
    type: 'APPLICATION:RESET_USER',
  }
}
