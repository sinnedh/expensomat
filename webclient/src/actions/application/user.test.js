import * as actions from './user'

describe('APPLICATION', () => {
  const token = 'ABCD123'
  const user = {name: 'Keek', role: 'admin'}

  it('handles setToken', () => {
    const action = actions.setToken(token)
    expect(action).toMatchSnapshot()
  })

  it('handles setUser', () => {
    const action = actions.setUser(user)
    expect(action).toMatchSnapshot()
  })

  it('handles resetUser', () => {
    const action = actions.resetUser()
    expect(action).toMatchSnapshot()
  })
})
