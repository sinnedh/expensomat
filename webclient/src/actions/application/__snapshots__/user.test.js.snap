// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APPLICATION handles resetUser 1`] = `
Object {
  "type": "APPLICATION:RESET_USER",
}
`;

exports[`APPLICATION handles setToken 1`] = `
Object {
  "token": "ABCD123",
  "type": "APPLICATION:SET_TOKEN",
}
`;

exports[`APPLICATION handles setUser 1`] = `
Object {
  "type": "APPLICATION:SET_USER",
  "user": Object {
    "name": "Keek",
    "role": "admin",
  },
}
`;
